src/main/java/io/github/vcuswimlab/stackintheflow/model/L2H/L2HPredictor.java

```diff
@@ -0,0 +1,209 @@
+		data.prepareTopicCoherence(numTopWords);
+
+		testSampler = new CustomL2H();
+		testSampler.setVerbose(true);
+        testSampler.setDebug(false);
+        testSampler.setLog(false);
+        testSampler.setReport(false);
+        testSampler.configure("./output", data.getWordVocab().size(), alpha, beta, a0, b0,
+				new MSTBuilder(data.getLabels(), data.getLabelVocab()), false, false, AbstractSampler.InitialState.PRESET, PathAssumption.MAXIMAL, false, burnIn, modelIterations, sampleLag, 1);
+        testSampler.setTestConfigurations(burnIn, modelIterations, sampleLag);
+	}
+
+	public String computeL2HPredictions(String fileToPredict) throws Exception {
+		BufferedReader br;
+		br = new BufferedReader(new InputStreamReader(this.getClass().getClassLoader().getResourceAsStream(fileToPredict)));
+		Map<Integer, Integer> wordOccurrenceMap = new HashMap<>();
+		try {
+			String line;
+			while ((line = br.readLine()) != null) {
+				String[] parts = line.split("\\s+");
+				for(int i = 1; i < parts.length; i++) {
+					String name = parts[i-1] + " " + parts[i];
+					if(nameToIdMap.containsKey(name)) {
+						int index = nameToIdMap.get(name);
+						if(wordOccurrenceMap.containsKey(index)) {
+							wordOccurrenceMap.put(index, wordOccurrenceMap.get(index)+1);
+						} else {
+							wordOccurrenceMap.put(index, 1);
+						}
+					}
+				}
+			}
+		} finally {
+			br.close();
+		}
+
+		StringBuilder builder = new StringBuilder();
+		builder.append(wordOccurrenceMap.size() + " ");
+		for(Map.Entry<Integer,Integer> e : wordOccurrenceMap.entrySet()) {
+			builder.append(e.getKey() + ":" + e.getValue() + " ");
+		}
+
+		int[][] newWords = new TextDataLoader(new ByteArrayInputStream(builder.toString().getBytes())).getWords();
+		double[][] initPredictions = computeInitPredictions(newWords, data.getLabelVocab().size());
+
+		try {
+            return testSampler.sampleNewDocuments(stateFile, newWords, null, initPredictions, 20);
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new RuntimeException();
+        }
+	}
+
+	public List<TagPrediction> computeMostLikelyTags(InputStream stream, int topN) throws Exception {
+		BufferedReader br = new BufferedReader(new InputStreamReader(stream));
+		
+		TagPrediction[] tagProbabilities; 
+		try {
+			br.readLine();
+			String line = br.readLine();
+			String[] parts = line.split("\\s+");
+			tagProbabilities = new TagPrediction[parts.length-1];
+			for(int i = 0; i < parts.length-1; i++) {
+				tagProbabilities[i] = new TagPrediction(data.getLabelVocab().get(i),Double.parseDouble(parts[i]));
+			}
+		} finally {
+			br.close();
+		}
+
+		if(topN <= 0 || topN > tagProbabilities.length) {
+			topN = tagProbabilities.length;
+		}
+
+		List<TagPrediction> result = new ArrayList<>();
+		Arrays.sort(tagProbabilities);
+		for(int i = 0; i < topN; i++) {
+			result.add(tagProbabilities[i]);
+		}
+		return result;
+	}
+
+	public static void main(String[] args) throws Exception {
```

Comment:
Again, I'm still confused what place a `public static void main` method has anywhere in the plugin's source. Are we calling part of this plugin as a command line app at runtime?